packages/diagrams/backend/sirius-components-diagrams-layout/src/main/java/org/eclipse/sirius/components/diagrams/layout/ELKDiagramConverter.java

```diff
@@ -288,4 +292,87 @@ private void convertEdge(Edge edge, ElkNode elkDiagram, Map<String, ElkConnectab
 
         id2ElkGraphElements.put(edge.getId(), elkEdge);
     }
+
+    private double getMaxPadding(Node node, ISiriusWebLayoutConfigurator layoutConfigurator) {
+        double maxPadding = 0d;
+        ElkPadding nodeLabelsPadding = layoutConfigurator.configureByType(node.getType()).getProperty(CoreOptions.NODE_LABELS_PADDING);
+        double leftPadding = 0d;
+        double rightPadding = 0d;
+        double topPadding = 0d;
+        double bottomPadding = 0d;
+
+        leftPadding = nodeLabelsPadding.getLeft();
+        rightPadding = nodeLabelsPadding.getRight();
+        topPadding = nodeLabelsPadding.getTop();
+        bottomPadding = nodeLabelsPadding.getBottom();
+        maxPadding = Double.max(Double.max(leftPadding, rightPadding), Double.max(topPadding, bottomPadding));
+
+        return maxPadding;
+    }
+
+    private String getNodeLabelType(Node node, ISiriusWebLayoutConfigurator layoutConfigurator) {
+        String labelType = "label:";
+        String nodeType = node.getType();
+        EnumSet<NodeLabelPlacement> nodeLabelPlacementSet = layoutConfigurator.configureByType(nodeType).getProperty(CoreOptions.NODE_LABELS_PLACEMENT);
+        if (nodeLabelPlacementSet.stream().anyMatch(NodeLabelPlacement.OUTSIDE::equals)) {
+            labelType += "outside";
+            if (NodeType.NODE_IMAGE.equals(nodeType)) {
+                labelType += "-center";
+            }
+        } else if (NodeType.NODE_RECTANGLE.equals(nodeType) && !this.hasHeader(node)) {
+            labelType += "inside";
+            String verticalAlignment = this.getVerticalAlignment(nodeLabelPlacementSet);
+            labelType += verticalAlignment;
+
+            String horizontalAlignment = this.getHorizontalAlignment(nodeLabelPlacementSet);
+            labelType += horizontalAlignment;
+        } else {
+            labelType += "inside-center";
+        }
+
+        return labelType;
+    }
+
+    private String getVerticalAlignment(EnumSet<NodeLabelPlacement> nodeLabelPlacementSet) {
+        String verticalAlignment = "-v_center";
+        for (NodeLabelPlacement nodeLabelPlacement : nodeLabelPlacementSet) {
```

Comment:
It feels strange to iterate on a set for this, but at least `EnumSet` guarantees a predictable iteration order...

Reply:
I've updated this method and move it in a new dedicated service class (ELKPropertiesService.java).